server/storage/database.ts

```diff
@@ -8,10 +8,33 @@ import Model from "@server/models/base/Model";
 import Logger from "../logging/Logger";
 import * as models from "../models";
 
+/**
+ * Returns the effective database URL, either from DATABASE_URL or constructed
+ * from individual database components.
+ */
+function getEffectiveDatabaseUrl(): string {
```

Comment:
Instead of constructing the url just pass these options directly into the `Sequelize` constructor